### PR TITLE
IsEnum

### PR DESCRIPTION
**Pull request by FBruzzesi**
In a similar fashion to #68, it introduces three types to check enum's:

- `IsEnumType`: checks if a class definition is subclass of `Enum`, and possibly checks exact matching of member values using `IsStrictDict`.
- `IsPartialEnumType`: checks if a class definition is subclass of `Enum`, and possibly checks partial matching of member values using `IsPartialDict`.
- `IsEnum`: checks if an instance is from `Enum` or subclass (arguably this is the most "WIP", since I think it should be possible to check if a value is part of an enum members)

I am tagging this as WIP as we didn't discuss about it.
(Worked on this as it is in the list of #3 )

### CODE REVIEW

File: dirty_equals/_other.py [hunk 1]
```diff
@@ -369,3 +371,125 @@ def equals(self, other: Any) -> bool:
...
+
+    def __init__(self, enum_cls: Type[Enum] = Enum):
+        """
+        Args:
+            enum_cls: Enum class to check against.
+        """
+        self._enum_cls = enum_cls
+
+    def equals(self, other: Any) -> bool:
+        return isinstance(other, self._enum_cls)
```
Comment (FBruzzesi) [inline dirty_equals/_other.py:405]: One idea to account for checking if `other` is a value in the `_enum_cls` could be:

```python
if isinstance(other, Enum):
    return isinstance(other, self._enum_cls)
else:
    return other in {i.value for i in self._enum_cls}
```

**Comment by FBruzzesi** [comment]
Adjusted with feedbacks from #68, still think that comment for `IsEnum` is relevant and worth discussion.

**Comment by FBruzzesi** [comment]
@samuelcolvin I completely forgot about this PR and deleted the fork.
Not sure if you have the time to look into it and/or interested in having such feature

**Comment by samuelcolvin** [comment]
Hi, sorry for the slow reply, life has been fairly busy.

Yes, I think it's worth adding. Feel free to create a new PR.

**Comment by FBruzzesi** [comment]
> Hi, sorry for the slow reply, 

Absolutely no need to be sorry, I should have been the one not deleting the PR 🙈

> life has been fairly busy.

I can barely imagine!
 
> Yes, I think it's worth adding. Feel free to create a new PR.

✅

**Pull request closed without merging**